src/components/Notifications/Notifier.js

```diff
@@ -1,40 +1,44 @@
+/* istanbul ignore file */
 import { Component } from "react";
 import PropTypes from "prop-types";
 import { connect } from "react-redux";
 import { withSnackbar } from "notistack";
 import { removeSnackbar } from "../../actions/notificationActions";
 
 class Notifier extends Component {
-    state = {
-        displayed: [],
-    };
+    displayed = [];
 
-    storeDisplayed = (key) => {
-        this.setState(({ displayed }) => ({
-            displayed: [...displayed, key],
-        }));
+    storeDisplayed = (id) => {
+        this.displayed = [...this.displayed, id];
     };
 
-    render() {
-        const { notifications, enqueueSnackbar, removeSnackbar } = this.props;
-        const { displayed } = this.state;
+    shouldComponentUpdate({ notifications: newSnacks = [] }) {
+        const { notifications: currentSnacks } = this.props;
+        for (let i = 0; i < newSnacks.length; i += 1) {
+            if (!currentSnacks.filter(({ key }) => newSnacks[i].key === key).length)
```

Comment:
A plain ES6 `Set`? No. (https://github.com/reduxjs/redux/issues/1499) I tried it in another project and it was not a good idea. An alternative mentioned in the Redux docs is `Immutable.js` (https://redux.js.org/recipes/using-immutablejs-with-redux), but it presents a lot of pitfalls and requires careful programming because its structures are handled differently than normal JS Objects, so you need to know what you're doing basically. And, even if I know that me, or you, or 1 or 2 more contributors would be careful as well, and do their homework of reading the pitfalls and the How-To's, the rest won't, which will cause a lot of problems and anti-patterns, leading to inefficient solutions, inadvertently, and, as they say in the link, Immutable.js is hard to debug, because they are not normal Objects. So, I think I prefer this O(n^2).

Reply:
Fair point, did not know about that. Thought that the ES6 `Set` could be used as a drop-in replacement.

As such, disregard the initial comment.